src/helpers/feed.ts

```diff
@@ -0,0 +1,30 @@
+import { HandleResponse, castResponse } from './response';
+import { isDefined } from './typescript';
+
+const TOTAL_RESPONSE_HEADER = 'x-total';
+const getTotalFromApiFeedResponse = (response: Response) => {
+  const totalsStr = response.headers.get(TOTAL_RESPONSE_HEADER);
+  if (isDefined(totalsStr)) {
+    const total = parseInt(totalsStr);
+    if (Number.isInteger(total)) {
+      return total;
+    } else {
+      throw new Error('Expected x-total header to be valid integer.');
```

Comment:
What if the API throws a 503 error? Will this error get returned before the API error? Because it will be confusing to a user if this error gets returned before the 5xx error, as they'll be wrongly debugging the wrong error

Reply:
This function is only reached if `response.ok === true`, i.e. if the status code is in the 2xx range. So it won't override any APi errors.